src/Microsoft.AspNetCore.Razor.Evolution/ITagHelperBoundAttributeDescriptorBuilder.cs

```diff
@@ -116,15 +116,7 @@ public ITagHelperBoundAttributeDescriptorBuilder AddDiagnostic(RazorDiagnostic d
 
         public BoundAttributeDescriptor Build()
         {
-            var validationDiagnostics = Validate();
-
-            var diagnostics = new List<RazorDiagnostic>();
-            diagnostics.AddRange(validationDiagnostics);
-
-            if (_diagnostics != null)
-            {
-                diagnostics.AddRange(_diagnostics);
-            }
+            Validate();
```

Comment:
This is wrong. These errors are **computed** based on the state of builder when `Build` is called. Storing them is always wrong.